### PR TITLE
5357: [TEST]: Add cleanup to make tests Tidy

### PR DESCRIPTION
**Pull request by yuliiamir**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy [hunk 1]
```diff
@@ -285,12 +295,7 @@ class SwitchValidationSingleSwFlowSpec extends HealthCheckSpecification {
 
         cleanup:
         flow && !deleteFlow && flowHelperV2.deleteFlow(flow.flowId)
-        if (!testIsCompleted) {
-            northbound.synchronizeSwitch(sw.dpId, true)
-            Wrappers.wait(RULES_INSTALLATION_TIME) {
-                assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
-            }
-        }
+        flow && !testIsCompleted && synchronizeAndValidateRules(sw)
```
Comment (pkazlenka) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy:298]: You don't need flow && here, I guess.

**Review by pkazlenka — CHANGES_REQUESTED** [review]
Mostly good, just move synchronization actions to helper method (if not exists yet).

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy (hunk 1 above)
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy:298]: There is no need to sync the switch when no flow has been added.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy [hunk 2]
```diff
@@ -208,14 +212,19 @@ class SwitchValidationSingleSwFlowSpec extends HealthCheckSpecification {
...
         Wrappers.wait(WAIT_OFFSET) {
             def switchValidateInfoAfterDelete = northbound.validateSwitch(sw.dpId)
             switchValidateInfoAfterDelete.verifyRuleSectionsAreEmpty()
             switchValidateInfoAfterDelete.verifyMeterSectionsAreEmpty()
         }
+        def testIsCompleted = true
+
+        cleanup:
+        flow && !deletedFlow && flowHelperV2.deleteFlow(flow.flowId)
+        flow && !testIsCompleted && synchronizeAndValidateRules(sw)
```
[the remark this answers is not in the document]
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy:227]: There is no need to delete flow or do switch sync when no flow has been added.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy [hunk 3]
```diff
@@ -284,6 +285,8 @@ class SwitchSyncSpec extends BaseSpecification {
 
         cleanup: "Delete the flow"
         flow && flowHelperV2.deleteFlow(flow.flowId)
+        involvedSwitches.each {
+            !northbound.validateSwitch(it.dpId).meters.misconfigured.isEmpty() && northbound.synchronizeSwitch(it.dpId, true) }
```
[the remark this answers is not in the document]
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy]: reused existing method

**Review by pkazlenka — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Pull request merged**